src/jit/compiler.h

```diff
@@ -1329,11 +1374,11 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
     DWORD expensiveDebugCheckLevel;
 #endif
 
-#if FEATURE_MULTIREG_RET
+#if FEATURE_MULTIREG_RET 
     GenTreePtr               impAssignStructClassToVar(GenTreePtr op, CORINFO_CLASS_HANDLE hClass);
 #endif // FEATURE_MULTIREG_RET
 
-#ifdef _TARGET_ARM_
+#ifdef FEATURE_HFA
```

Comment:
Does all these routines need to be under #ifdef FEATURE_HFA?   In view of avoiding #ifdef'ed code as much as possible, i would suggest the following:


IsHfa(hClass) and IsHfa(GenTreePtr) - since Jit-EE method is defined to return false for non-hfa targets, these methods don't need to be inside FEATURE_HFA.

GetHfaType() - these routines will return TYP_UNDEF
GetHfaSlots() - in turn is calling GetHfaSlots(hClass) that asserts that hClass is known to be a HFA.  #ifdef could be moved inside the routine with #else being unreached().

GetHfaType() - this will return TYP_UNDEF if the type is not hfa.
GetHfaSlots(hClass) - #ifdef should be moved inside the routine with #else being unreached.